generators/client/templates/src/main/webapp/app/blocks/config/_http.config.js

```diff
@@ -19,7 +19,7 @@
         $urlRouterProvider.otherwise('/');
 
         $httpProvider.interceptors.push('errorHandlerInterceptor');
-        $httpProvider.interceptors.push('authExpiredInterceptor');<% if (authenticationType == 'oauth2' || authenticationType == 'jwt') { %>
+        $httpProvider.interceptors.push('authExpiredInterceptor');<% if (authenticationType == 'oauth2' || authenticationType == 'jwt' || applicationType == 'gateway') { %>
```

Comment:
same here `this.applicationType == 'gateway'` check is not required as a gateway will always have `authenticationType == 'jwt'` and hence it will always be true here